### PR TITLE
Created Resources implementation

### PR DESCRIPTION
**Pull request by umbreak**
Fixes: https://github.com/BlueBrain/nexus/issues/1521

- Added DB serialization for `ResourceEvent`.
- Added Encoder/Decoder for `ResourceRef` (that required some additional methods in `Iri`).
- Small fixes on decoding (forcing strict decoding for input payload from the client only).

### CODE REVIEW

**Review by imsdu — CHANGES_REQUESTED** [review]

File: delta/sdk/src/main/scala/ch/epfl/bluebrain/nexus/delta/sdk/model/ResourceRef.scala [hunk 1]
```diff
@@ -23,24 +31,49 @@ object ResourceRef {
...
     * A reference annotated with a tag.
     *
-    * @param iri the reference identifier as an iri
-    * @param tag the reference tag
+    * @param original the original iri
+    * @param iri      the reference identifier as an iri
+    * @param tag      the reference tag
     */
-  final case class Tag(iri: Iri, tag: String) extends ResourceRef
+  final case class Tag(original: Iri, iri: Iri, tag: String) extends ResourceRef
```
Comment (imsdu) [inline delta/sdk/src/main/scala/ch/epfl/bluebrain/nexus/delta/sdk/model/ResourceRef.scala:54]: tag is a label too here no ?

File: delta/sdk/src/main/scala/ch/epfl/bluebrain/nexus/delta/sdk/model/ResourceRef.scala [hunk 2]
```diff
@@ -13,7 +16,12 @@ sealed trait ResourceRef extends Product with Serializable {
     */
   def iri: Iri
```
Comment (imsdu) [inline delta/sdk/src/main/scala/ch/epfl/bluebrain/nexus/delta/sdk/model/ResourceRef.scala:17]: It is not clear when reading the comments the difference between an the iri and the original

**Review by imsdu — CHANGES_REQUESTED** [review]

File: delta/rdf/src/main/scala/ch/epfl/bluebrain/nexus/delta/rdf/IriOrBNode.scala [hunk 3]
```diff
@@ -47,6 +48,51 @@ object IriOrBNode {
...
+      */
+    def removeQueryParams(keys: String*): Iri =
+      if (rawQuery().isEmpty) this
+      else queryParams(Query(query().toMap -- keys))
+
+    /**
+      * Override the current query parameters with the passed ones
+      */
+    def queryParams(query: Query): Iri = {
+      val sb = new StringBuilder
```
Comment (imsdu) [inline delta/rdf/src/main/scala/ch/epfl/bluebrain/nexus/delta/rdf/IriOrBNode.scala:75]: The iriFactory has methods that could help here

**Review by imsdu — CHANGES_REQUESTED** [review]

File: delta/rdf/src/main/scala/ch/epfl/bluebrain/nexus/delta/rdf/IriOrBNode.scala (hunk 3 above)
Comment (umbreak) [reply delta/rdf/src/main/scala/ch/epfl/bluebrain/nexus/delta/rdf/IriOrBNode.scala:75]: The construct method does not work, it is [not implemented](https://github.com/apache/jena/blob/master/jena-iri/src/main/java/org/apache/jena/iri/IRIFactory.java#L473)

**Review by imsdu — APPROVED** [review]

**Review by bogdanromanx — APPROVED** [review]

**Pull request merged**